packages/test/generatedCode/models/microsoft/graph/attachmentImpl.ts

```diff
@@ -0,0 +1,64 @@
+import {Attachment} from './attachment';
+import {EntityImpl} from './index';
+import {Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';
+
+export class AttachmentImpl extends EntityImpl implements Attachment, Parsable {
+    /** The MIME type. */
+    public contentType?: string | undefined;
+    /** true if the attachment is an inline attachment; otherwise, false. */
+    public isInline?: boolean | undefined;
+    /** The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is 2014-01-01T00:00:00Z */
+    public lastModifiedDateTime?: Date | undefined;
+    /** The attachment's file name. */
+    public name?: string | undefined;
+    /** The length of the attachment in bytes. */
+    public size?: number | undefined;
+    /**
+     * Instantiates a new attachment and sets the default values.
+     * @param attachmentParameterValue 
+     */
+    public constructor(attachmentParameterValue?: Attachment | undefined) {
+        super();
```

Comment:
shouldn't the parameter be passed to the parent constructor as well to get the properties from the parent assigned as well?

Reply:
I can see that you updated to the following.

```typescript
super({
            additionalData: extensionParameterValue?.additionalData,
            id: extensionParameterValue?.id,
        });
```

Why not do that instead 

```
super(extensionParameterValue);
```

Since the derived type will fulfill the base type interface? (to avoid generating too much code)